alert-database/src/test/java/com/synopsys/integration/alert/database/api/DefaultNotificationAccessorTest.java

```diff
@@ -325,6 +326,24 @@ public void setNotificationsProcessedTest() {
         Mockito.verify(notificationContentRepository).saveAll(Mockito.any());
     }
 
+    @Test
+    public void setNotificationsProcessedByIdTest() {
+        NotificationEntity notificationEntity = new NotificationEntity(id, DateUtils.createCurrentDateTimestamp(), provider, providerConfigId, DateUtils.createCurrentDateTimestamp(), notificationType, content, false);
+        ConfigurationModel configurationModel = createConfigurationModel();
+        Set<Long> notificationIds = Set.of(1L);
+
+        NotificationContentRepository notificationContentRepository = Mockito.mock(NotificationContentRepository.class);
+        ConfigurationAccessor configurationAccessor = Mockito.mock(ConfigurationAccessor.class);
+
+        Mockito.when(notificationContentRepository.findAllById(Mockito.any())).thenReturn(List.of(notificationEntity));
+        Mockito.when(configurationAccessor.getConfigurationById(Mockito.any())).thenReturn(Optional.of(configurationModel));
+
+        DefaultNotificationAccessor notificationManager = new DefaultNotificationAccessor(notificationContentRepository, null, null, configurationAccessor);
+        notificationManager.setNotificationsProcessedById(notificationIds);
+
+        Mockito.verify(notificationContentRepository).saveAll(Mockito.any());
```

Comment:
There is no check to see if the "processed" value was changed?

Reply:
Because its the noificationContentRepository is mocked we don't have a way of checking the processed value similar to the other test above. I created the IT tests in order to check if the value is set to processed or not.